src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package com.synacy.lesson02.exercises;
+
+import com.synacy.lesson02.exercises.domain.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.xml.ws.soap.MTOM;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+
+    private EnrollmentService enrollmentService;
+    private EnrollmentDto enrollmentDetails;
+
+    @Mock StudentProfileService studentProfileService;
+    @Mock CourseClassService courseClassService;
+    @Mock EnrollmentNotificationService enrollmentNotificationService;
+    @Mock StudyLoadFormatter studyLoadFormatter;
+    @Mock SystemService systemService;
+
+    @Before
+    public void setup(){
+        enrollmentService = new EnrollmentService();
+
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+        enrollmentService.setSystemService(systemService);
+        enrollmentDetails = mock(EnrollmentDto.class);
```

Comment:
Let's refrain mocking domain models in the setup method. We want our test methods as isolated from each other as possible. With that in mind, all domain models should be mocked within the test method that requires it. In this case, each test method should have a mocked `EnrollmentDto` that is unique among each other.